src/Cake/Scripting/Roslyn/RoslynScriptEngine.cs

```diff
@@ -2,56 +2,33 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#if !NETCORE
-using System;
 using System.Collections.Generic;
 using Cake.Core.Diagnostics;
+using Cake.Core.Reflection;
 using Cake.Core.Scripting;
-using Cake.Scripting.Roslyn.Nightly;
-using Cake.Scripting.Roslyn.Stable;
 
 namespace Cake.Scripting.Roslyn
 {
     internal sealed class RoslynScriptEngine : IScriptEngine
     {
-        private readonly RoslynScriptSessionFactory _stableFactory;
         private readonly CakeOptions _options;
-        private readonly RoslynNightlyScriptSessionFactory _nightlyFactory;
+        private readonly IAssemblyLoader _loader;
         private readonly ICakeLog _log;
 
-        public RoslynScriptEngine(
-            CakeOptions options,
-            RoslynScriptSessionFactory stableFactory,
-            RoslynNightlyScriptSessionFactory nightlyFactory,
-            ICakeLog log)
+        public RoslynScriptEngine(CakeOptions options, IAssemblyLoader loader, ICakeLog log)
         {
             _options = options;
-            _nightlyFactory = nightlyFactory;
-            _stableFactory = stableFactory;
+            _loader = loader;
             _log = log;
         }
 
         public IScriptSession CreateSession(IScriptHost host, IDictionary<string, string> arguments)
```

Comment:
While we are at it breaking stuff. Maybe we should remove `arguments` from the interface?